Components/Pages/Goals.js

```diff
@@ -72,24 +94,20 @@ const Goals = ({ navigation }) => {
     setNotification({ notification: notification });
   };
 
-  const sendPushNotification = async () => {
-    const message = {
-      to: expoPushToken,
-      sound: 'default',
-      title: 'Original Title',
-      body: 'And here is the body!',
-      data: { data: 'goes here' },
-      _displayInForeground: true,
-    };
-    const response = await fetch('https://exp.host/--/api/v2/push/send', {
-      method: 'POST',
-      headers: {
-        Accept: 'application/json',
-        'Accept-encoding': 'gzip, deflate',
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(message),
-    });
+  const attachToken = async () => {
+    if (!notificationsEnabled) {
```

Comment:
I wanted to do a check to determine if the request needs to be sent to the endpoint, but I don't think this is the right way to go necessarily. I more so want to determine if their push token was already stored previously so unnecessary requests aren't getting sent. Lmk if anyone has any suggestions.

Reply:
my idea is to use async storage. something like the below

  AsyncStorage.getItem("expoPushToken").then(token => {
    setnotificationsEnabled(token !== null);
  });


        await fetch(NOTIFICATION_URI, {
          method: "POST",
          headers: {
            "Accept": "application/json",
            "Accept-encoding": "gzip, deflate",
            "Content-Type": "application/json",
          },
          body: { expoPushToken },
        }).then(response => {
          AsyncStorage.setItem("expoPushToken", expoPushToken);
        });


AsyncStorage.setItem("notification_token",expoPushToken );